ui/dashboards/src/components/PanelDrawer/PanelSpecEditor.tsx

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ErrorAlert, JSONEditor } from '@perses-dev/components';
+import { PanelDefinition, QueryDefinition, UnknownSpec } from '@perses-dev/core';
+import {
+  usePlugin,
+  PanelPlugin,
+  OptionsEditorTabsProps,
+  TimeSeriesQueryEditor,
+  OptionsEditorTabs,
+} from '@perses-dev/plugin-system';
+
+export interface PanelSpecEditorProps {
+  panelDefinition: PanelDefinition;
+  onQueriesChange: (queries: QueryDefinition[]) => void;
```

Comment:
I kinda wonder if there should just be one:
`onPanelDefinitionChange` ?

Reply:
So that was my initial approach, but then I was running into issues with syncing the different changes happening in `PanelEditorForm`. For instance, showing/hiding the query editor when switching the panel types requires additional logic to save the last used queries. Overall i find this approach much more clear